opentnsim/strategy.py

```diff
@@ -278,10 +275,17 @@ def get_upperbound_for_power2v(vessel, width, depth, margin=0, bounds=(0, 20)):
     range."""
 
     # estimate the grounding velocity
-    grounding_v, depth, depth_squat, z_computed, margin = get_v(vessel, width, depth, margin=0, bounds=bounds)
+    # here we optionally try to take sinkage into account and try to compute the maximum velocity where we still have underkeel clearance
+    grounding_v, depth, margin = get_v(vessel, width, depth, margin=0, bounds=bounds)
 
+
+    # The next step is to compute the maximum power for the veloctiy.
+    # Here (for some reason we don't use a solver )
     # find max power velocity
-    velocity = np.linspace(0.01, grounding_v, 100)
+    velocity = np.linspace(0.01, grounding_v, 500)
```

Comment:
This should be improved (use scipy optimize, or even better an analytical computation). But that requires quite a bit of testing / effort. So for now I left it like this.